### DISCUSSION TITLE
How to link to global via relationship field?

### DISCUSSION THREAD

**Discussion by christian-reichart** [payloadcms/payload#2100] [Q&A]
Gday,
I have the following problem:

I have a collection of content pages and a global called "startpage", which is basically a content page, but special. 
I also have a link field, where users can select a page for internal linking.

How would I go about an internal link to the startpage, since globals can't be used in the relationship field?

Thanks for any hints!

**Comment by JessRynkar** [comment]
Hi @christian-reichart,

If you take a look at the link field group from our demo [here](https://github.com/payloadcms/public-demo/blob/master/src/fields/link.ts), you could follow this structure and use the `customURL` to manually point to your start page global.

Or if you need to return the full data from the start page, you could conditionally render another field that uses a hook to fetch the data. What works best will depend on your use case, there are more ways to do this out there!

**Reply by JessRynkar** [reply]
Also to keep it really simple you could just add a placeholder page for your homepage. Again depends on what you need here.

**Reply by christian-reichart** [reply]
I don't want to use a text field for the url, because I use the React Link component for internal linking and an anchor for custom/external url. I also dynamically set /de or /en in the frontend, so a hard coded link is not ideal. I ended up having a select field with the options internal, external and startpage. In hindsight, my startpage should have just been a regular content page, maybe I can refactor this in the future. Thanks for all the tips!

**Comment by PP-Tom** [comment]
This would be a really nice addition, there is many reasons to have a relationship to a global other than just the URL. In my case we are wanting to get what projects are featured on that page.

**Comment by robinscholz** [comment]
We just ran into this problem as well. This seems like a huge oversight in payloads logic. Globals are perfect for single pages and being able to refer them anywhere in a relationship field seems like a basic requirement.

**Comment by robinscholz** [comment]
It would be great if Payload V3 could include this!

**Comment by studiodrimmel** [comment]
I don't know if this would help anyone here, but it might help other people who are having this issue and find this discussion via Google as I was facing the same problem, and I ended up adding a different collection to reference the slugs of my Globals.

This makes it easy to update url's across the application in stead of hardcoding them in a url field. If the url changes you only have to replace them in this collection, in stead of searching for them in your entire website. 

```typescript
const GlobalPages: CollectionConfig = {
  slug: 'globalPages',
  admin: {
    useAsTitle: 'name',
  },
  access: {
    read: () => true,
  },
  fields: [
    {
      name: 'name',
      type: 'text',
      required: true,
    },
    {
      name: 'slug',
      type: 'text',
      required: true,
    },
  ],
}

export default GlobalPages
```

```typescript
{
      name: 'reference',
      label: 'Document to link to',
      type: 'relationship',
      relationTo: ['pages', 'globalPages'],
      required,
      admin: {
        condition: (_, siblingData) => siblingData?.type === 'reference',
      },
    },
```

**Reply by mbret** [reply]
nice, it worked elegantly. Going with this solution

**Reply by PP-Tom** [reply]
This is a great solution, didn't think of this. I'm going with this one!

If it's just for linking this works perfectly, relationships to the global would still be nice in some use-cases.

**Comment by vanska** [comment]
I've built a plugin that keeps track of document references per collection and protects documents from deletion when still being referenced while alerting users of existing references. Not being able to reference globals makes globals unusable for this use case.

## Example

Global `header` references a `pages` document in a required field. We don't want user to delete these referenced documents as the application layer might expect data to be there.

<img width="722" height="351" alt="image" src="https://github.com/user-attachments/assets/475b3c20-8f2b-4e9c-9b34-c88442c4ef6b" />

Somewhat related issues:
- https://github.com/payloadcms/payload/discussions/4650
- https://github.com/payloadcms/payload/issues/11177
- https://github.com/payloadcms/payload/discussions/558
- https://github.com/payloadcms/payload/discussions/15280

**Comment by ynamite** [comment]
+1
